src/tools/snapshot.ts

```diff
@@ -198,14 +202,34 @@ const screenshot: Tool = {
   handle: async (context, params) => {
     const validatedParams = screenshotSchema.parse(params);
     const tab = context.currentTab();
-    const options: playwright.PageScreenshotOptions = validatedParams.raw ? { type: 'png', scale: 'css' } : { type: 'jpeg', quality: 50, scale: 'css' };
-    const screenshot = await tab.page.screenshot(options);
+    const fileName = path.join(os.tmpdir(), sanitizeForFilePath(`page-${new Date().toISOString()}`)) + `${validatedParams.raw ? '.png' : '.jpeg'}`;
+    const options: playwright.PageScreenshotOptions = validatedParams.raw ? { type: 'png', scale: 'css', path: fileName } : { type: 'jpeg', quality: 50, scale: 'css', path: fileName };
+    if (validatedParams.wholePage) {
+      await tab.page.screenshot(options);
```

Comment:
This needs proper code generated. Like

```js
        const code = [
          `// Save screenshot as ${fileName}`,
          `await page.screenshot(${javascript.formatObject(options)});`
        ];
        return { code };
```

Also move into runAndWaitWithSnapshot to emit updated snapshot.